wcfsetup/install/files/ts/WoltLabSuite/Core/Form/Builder/Field/Dependency/Manager.ts

```diff
@@ -0,0 +1,303 @@
+
+    EventHandler.fire("com.woltlab.wcf.form.builder.dependency", "checkContainers");
+
+    // finish checking containers and check if containters should be checked again
+    _checkingContainers = false;
+    if (_checkContainersAgain) {
+      this.checkContainers();
+    }
+  },
+
+  /**
+   * Checks if all dependencies are met.
+   */
+  checkDependencies(): void {
+    const obsoleteNodeIds: string[] = [];
+
+    _nodeDependencies.forEach((nodeDependencies, nodeId) => {
+      const dependentNode = document.getElementById(nodeId);
+      if (dependentNode === null) {
+        obsoleteNodeIds.push(nodeId);
+
+        return;
+      }
+
+      let dependenciesMet = true;
+      nodeDependencies.forEach((dependency) => {
+        if (!dependency.checkDependency()) {
+          dependentNode.style.display = "none";
+          dependenciesMet = false;
+        }
+      });
+
+      if (dependenciesMet) {
+        this._show(dependentNode);
+      }
+    });
+
+    obsoleteNodeIds.forEach((id) => _nodeDependencies.delete(id));
+
+    this.checkContainers();
+  },
+
+  /**
+   * Returns `true` if the given node has been hidden because of its own dependencies.
+   */
+  isHiddenByDependencies(node: HTMLElement): boolean {
+    if (_dependencyHiddenNodes.has(node)) {
+      return true;
+    }
+
+    let returnValue = false;
+    _dependencyHiddenNodes.forEach((hiddenNode) => {
+      if (node.contains(hiddenNode)) {
+        returnValue = true;
+      }
+    });
+
+    return returnValue;
+  },
+
+  /**
+   * Registers the form with the given id with the dependency manager.
+   */
+  register(formId: string): void {
+    const form = document.getElementById(formId);
+
+    if (form === null) {
+      throw new Error("Unknown element with id '" + formId + "'");
+    }
+
+    if (_forms.has(form)) {
+      throw new Error("Form with id '" + formId + "' has already been registered.");
+    }
+
+    _forms.add(form);
+  },
+
+  /**
+   * Unregisters the form with the given id and all of its dependencies.
+   */
+  unregister(formId: string): void {
+    const form = document.getElementById(formId);
+
+    if (form === null) {
+      throw new Error("Unknown element with id '" + formId + "'");
+    }
+
+    if (!_forms.has(form)) {
+      throw new Error("Form with id '" + formId + "' has not been registered.");
+    }
+
+    _forms.delete(form);
+
+    _dependencyHiddenNodes.forEach((hiddenNode) => {
+      if (form.contains(hiddenNode)) {
+        _dependencyHiddenNodes.delete(hiddenNode);
+      }
+    });
+    _nodeDependencies.forEach((dependencies, nodeId) => {
+      if (form.contains(document.getElementById(nodeId))) {
+        _nodeDependencies.delete(nodeId);
+      }
+
+      dependencies.forEach((dependency) => {
+        dependency.getFields().forEach((field) => {
+          _fields.delete(field.id);
+
+          _validatedFieldProperties.delete(field);
+        });
+      });
+    });
+  },
+};
+
+Core.enableLegacyInheritance(Manager);
```

Comment:
Manager is not a class. It probably should be exported as bare methods unless there are compatibility reasons preventing that.